### PR TITLE
[PropertyInfo] Fix multi phpdoc covered promoted properties

### PR DESCRIPTION
**Pull request by simPod**
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | yes
| Tickets       | Fix #46550
| License       | MIT

I came up with this solution

### CODE REVIEW

File: src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php [hunk 1]
```diff
@@ -99,6 +99,14 @@ public function getTypes(string $class, string $property, array $context = []):
                 continue;
             }
 
+            if (
+                $tagDocNode->value instanceof ParamTagValueNode
+                && $prefix === null
```
Comment (nicolas-grekas) [inline src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php]: yoda rulez

File: src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php [hunk 2]
```diff
@@ -99,6 +99,14 @@ public function getTypes(string $class, string $property, array $context = []):
...
+                && $tagDocNode->value->parameterName !== '$' . $property
```
Comment (nicolas-grekas) [inline src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php]: no spaces around the `.`

**Review by nicolas-grekas — COMMENTED** [review]
/cc @ostrolucky in case you can come up with another failing test case? or is this robust enough?

**Comment by ostrolucky** [comment]
With this, maybe my patch at https://github.com/symfony/symfony/pull/46453 is not necessary anymore?

File: src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php (hunk 1 above)
Comment (simPod) [reply src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php]: Btw what is the rationale behind using it here in php project?

**Comment by ostrolucky** [comment]
Going to answer my own comment
```php
if (self::MUTATOR === $source && !$this->filterDocBlockParams($phpDocNode, $property)) {
    return null;
}
```

this guard can now indeed by removed, since you added more generic solution.

**Comment by simPod** [comment]
Indeed, cool

**Review by ostrolucky — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @simPod.

**Comment by nicolas-grekas** [comment]
And thank you @ostrolucky also.

**Pull request merged**